vector/src/main/java/im/vector/app/features/ftue/FTUEVariantFactory.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.ftue
+
+import im.vector.app.features.VectorFeatures
+import im.vector.app.features.login.LoginViewModel
+import im.vector.app.features.login2.LoginViewModel2
+import javax.inject.Inject
+
+class FTUEVariantFactory @Inject constructor(
+        private val vectorFeatures: VectorFeatures,
+) {
+
+    fun create(activity: FTUEActivity, loginViewModel: Lazy<LoginViewModel>, loginViewModel2: Lazy<LoginViewModel2>) = when (vectorFeatures.loginVariant()) {
```

Comment:
these separate FTUE and FTUE_WIP variants only need to exist whilst the changes are ported over and updated from the WIP